File: competitive-events-api/src/main/java/es/jocasolo/competitiveeventsapi/dto/user/UserDTO.java
package es.jocasolo.competitiveeventsapi.dto.user;

import java.io.Serializable;
import java.util.Date;

import es.jocasolo.competitiveeventsapi.dto.DTO;
import es.jocasolo.competitiveeventsapi.dto.image.ImageDTO;
import lombok.Getter;
import lombok.Setter;

public class UserDTO extends DTO implements Serializable {

	private static final long serialVersionUID = 1L;

	@Getter
	@Setter
	private String id;

	@Getter
	@Setter
	private String name;

	@Getter
	@Setter
	private String surname;

	@Getter
	@Setter
	private String description;

	@Getter
	@Setter
	private ImageDTO avatar;

	@Getter
	@Setter
	private Date registerDate;

	@Getter
	@Setter
	private Date birthDate;

	@Override
	public String toString() {
		return "UserDTO [id=" + id + "]";
	}

}

File: competitive-events-app/app/src/main/java/es/jocasolo/competitiveeventsapp/dto/user/UserDTO.kt
package es.jocasolo.competitiveeventsapp.dto.user

import com.google.gson.annotations.SerializedName
import es.jocasolo.competitiveeventsapp.dto.image.ImageDTO
import java.util.*

open class UserDTO(
    @SerializedName("id") var id: String? = null,
    @SerializedName("name") var name: String? = null,
    @SerializedName("surname") var surname: String? = null,
    @SerializedName("description") var description: String? = null,
    @SerializedName("birthDate") var birthDate: Date? = null,
    @SerializedName("registerDate") var registerDate: Date? = null,
    @SerializedName("email") var email: String? = null,
    @SerializedName("avatar") var avatar: ImageDTO? = null
)